gm4_scuba_gear/data/scuba_gear/functions/main.mcfunction

```diff
@@ -0,0 +1,15 @@
+# check which SCUBA item is equipped
+execute as @a[tag=!gm4_in_water,gamemode=!spectator,nbt={Inventory:[{Slot:103b,tag:{gm4_scuba_gear:{item:"helmet"}}},{Slot:102b,tag:{gm4_scuba_gear:{item:"tank"}}}]}] at @s if block ~ ~1 ~ water run function scuba_gear:breathe
+execute as @a[tag=!gm4_in_water,gamemode=!spectator,nbt={Inventory:[{Slot:103b,tag:{gm4_scuba_gear:{item:"helmet"}}},{Slot:102b,tag:{gm4_scuba_gear:{item:"tank"}}}]}] at @s if block ~ ~1 ~ #scuba_gear:water_loggable[waterlogged=true] run function scuba_gear:breathe
+execute as @a[scores={gm4_sg_swim=1..},gamemode=!spectator,nbt={Inventory:[{Slot:100b,tag:{gm4_scuba_gear:{item:"flippers"}}}]}] run effect give @s dolphins_grace 1 0 true
```

Comment:
These 3 lines all check for a non spectator player with scuba gear, which should be optimized by running a function for each player that has a scuba gear item equipped.
```
execute as @a[gamemode=!spectator,nbt={Inventory:[{Slot:100b,tag:{gm4_scuba_gear:{}}}]}] run function scuba_gear:equipped
```